src/System.Net.Sockets/tests/FunctionalTests/UnixDomainSocketTest.cs

```diff
@@ -0,0 +1,221 @@
+        {
+            ManualResetEvent complete = (ManualResetEvent)args.UserToken;
+            complete.Set();
+        }
+
+        [Fact]
+        [PlatformSpecific(PlatformID.Windows)]
+        public void Socket_CreateUnixDomainSocket_Throws_OnWindows()
+        {
+            Assert.False(Capability.UnixDomainSocketsSupport());
+            Assert.Throws<SocketException>(() => new Socket(AddressFamily.Unix, SocketType.Stream, ProtocolType.Unspecified));
+        }
+
+        [Fact]
+        [PlatformSpecific(PlatformID.Linux | PlatformID.OSX)]
+        public void Socket_ConnectAsyncUnixDomainSocketEndPoint_Success()
+        {
+            Assert.True(Capability.UnixDomainSocketsSupport());
+
+            SocketTestServer server = SocketTestServer.SocketTestServerFactory(_endPoint, ProtocolType.Unspecified);
+
+            SocketAsyncEventArgs args = new SocketAsyncEventArgs();
+            args.RemoteEndPoint = _endPoint;
+            args.Completed += OnConnectAsyncCompleted;
+
+            ManualResetEvent complete = new ManualResetEvent(false);
+            args.UserToken = complete;
+
+            Socket sock = new Socket(AddressFamily.Unix, SocketType.Stream, ProtocolType.Unspecified);
+            Assert.True(sock.ConnectAsync(args));
+
+            complete.WaitOne();
+
+            Assert.Equal(SocketError.Success, args.SocketError);
+            Assert.Null(args.ConnectByNameError);
+
+            complete.Dispose();
+            sock.Dispose();
+            server.Dispose();
+        }
+
+        [Fact]
+        [PlatformSpecific(PlatformID.Linux | PlatformID.OSX)]
+        public void Socket_ConnectAsyncUnixDomainSocketEndPoint_NotServer()
+        {
+            Assert.True(Capability.UnixDomainSocketsSupport());
+
+            SocketAsyncEventArgs args = new SocketAsyncEventArgs();
+            args.RemoteEndPoint = _endPoint;
+            args.Completed += OnConnectAsyncCompleted;
+
+            ManualResetEvent complete = new ManualResetEvent(false);
+            args.UserToken = complete;
+
+            Socket sock = new Socket(AddressFamily.Unix, SocketType.Stream, ProtocolType.Unspecified);
+
+            bool willRaiseEvent = sock.ConnectAsync(args);
+            if (willRaiseEvent)
+            {
+                complete.WaitOne();
+            }
+
+            Assert.Equal(SocketError.SocketError, args.SocketError);
+
+            complete.Dispose();
+            sock.Dispose();
+        }
+
+        public void Dispose()
+        {
+            if (File.Exists(_path))
+            {
+                try
+                {
+                    File.Delete(_path);
+                }
+                catch
+                {
+                }
+            }
+
+        }
+
+        #region GC Finalizer test
+
+        // This test assumes sequential execution of tests and that it is going to be executed after other tests
+        // that used Sockets. 
```

Comment:
> if you want to test clean up after socket actions, you should include this verification in the same tests that exercise it,

AFAIK this is the long-term POR for the Sockets testbed, as the various instances of this pattern are no longer testing what they originally were in the first place. So yes, feel free to remove this test.